proposed/2021/TargetFrameworkAndPlatformsFilters.md

```diff
@@ -0,0 +1,196 @@
+  - .NET Standard
+- macCatalyst:
+  - .NET 6+
+    - `net6.0-maccatalyst`
+  - .NET Standard
+- tvOS:
+  - .NET 6+
+    - `net6.0-tvos`
+  - .NET Standard
+- Tizen:
+  - .NET 6+
+    - `net6.0-tizen`
+  - .NET Standard
+
+**Note:** .NET Standard is considered a special case that supports majority of these platforms depending on the [version of .NET standard](https://docs.microsoft.com/dotnet/standard/net-standard#net-implementation-support). If the framework precedence of a target framework supports .NET Standard, it's respective versions supported will be implicitly included.
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/SupportedPlatforms.png)
+
+#### Version Numbers
+
+There are many unique version numbers for all generations of .NET & supported platforms. The filtering experience should be smart enough to create a version list based on the current selection. Here are a few scenarios:
+
+- User selects `Any` for the .NET generation.
+  - Version filter populates all known versions for .NET Framework, .NET Core, .NET, and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 2.2, 3.0, 3.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8, 5.0, 6.0
+- User selects `Any` for the .NET generation and `Windows` for the supported platform.
+  - Version filter populates all known version for .NET Framework, .NET Core, .NET, and .NET Standard.
+  - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 2.2, 3.0, 3.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8, 5.0, 6.0
+- User selects `Any` for the .NET generation and `Android` for the supported platform.
+  - Version filter populates all known version for .NET and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 6.0
+- User selects `.NET Framework` for the .NET generation.
+  - Version filter populates all known version for .NET Framework and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8
+- User selects `.NET Framework` for the .NET generation and `Windows` for the supported platform.
+  - Version filter populates all known version for .NET Framework and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8
+- User selects `.NET Core` for the .NET generation and `macOS` for the supported platform.
+  - Version filter populates all known version for .NET Framework and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 2.2, 3.0, 3.1
+
+For further scenarios, see the [framework precedence](https://nugettools.azurewebsites.net/6.0.0-preview.3/framework-precedence) tool to understand how the version matrix would be calculated.
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/VersionNumbers.png)
+
+#### Complete Experience
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/CompleteExperience.png)
+
+## Search Syntax
```

Comment:
I would strongly recommend using query parameters instead of nuget.org's search syntax:

* Other search filters (semver level, prerelease, package type) are implemented using query parameters: https://docs.microsoft.com/nuget/api/search-query-service-resource#request-parameters
* nuget.org's search syntax is not part of the official NuGet protocol. Ideally Visual Studio uses the NuGet protocol and does not rely on APIs that are specific to nuget.org
* Visual Studio does not properly encode nuget.org's search syntax: https://github.com/NuGet/Home/issues/9043
* nuget.org's search syntax has a strange design that is highly inconsistent with other search syntaxes
*  It is easier for other server implementations to accept new query parameters then

Reply:
Should we support both? The search syntax NuGet.org provides is quite helpful & this would be a great addition.

Reply:
The search request would look something like this using query parameters: `?q=hello world&framework=net472&framework=net461`

Reply:
> Should we support both?

Perhaps. This would have the benefit of allowing VS users to filter packages before we update its UI.

Reply:
Can we support the same mechanism as GitHub for filters?

Reply:
Will selecting the UI filters also put the corresponding text filters in the search box? (same as GitHub)

Reply:
@anangaur That was an original hope. The problem is that NuGet.org does not support simple search syntax like `AND`. If we can remove that barrier, we could do a search syntax autocompletion in the textbox like GitHub.

Reply:
Resolving this feedback for now as these questions are more suited for the implementation. We definitely want ways to search by TFM in both the API & existing search syntax. The former being more important than the latter from the SDK perspective.